src/test/java/org/openrewrite/staticanalysis/NeedBracesTest.java

```diff
@@ -323,6 +333,177 @@ static void method() {
                       }
                       return;
                   }
+                  static void commentWhile(Object obj) {
+                      while (true) { // while comment
+                      }
+                      while (true) {
+                          return; // while comment with body
+                      }
+                      do {
+                          obj.notify(); // do while comment
+                      } while (true);
```

Comment:
Certainly understand choices have to be made at some point, but here I had sort of expected
```suggestion
                          obj.notify();
                      } while (true); // do while comment
```